.circleci/config.yml

```diff
@@ -0,0 +1,38 @@
+# Golang CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-go/ for more details
+version: 2
+jobs:
+  build_and_test:
+    docker:
+      - image: circleci/golang:1.13
+
+    working_directory: /go/src/github.com/cloudspannerecosystem/wrench
+
+    steps:
+      - checkout
+
+      - run:
+          name: set credentials
+          command: |
+            echo $GCLOUD_SERVICE_KEY > $HOME/gcloud-service-key.json
```

Comment:
I'm concerning that if we use service key like this it will be easily compromised by malicious users since everyone can run CI job. For example, everyone can send service key to any where in the CI job.  This is why I have not yet set up CI.
How do you think about this problem?

Reply:
This is a very valid concern, but CircleCI tries to ameliorate the risk through the "Pass secrets to builds from forked pull requests" setting in the "Advanced" section. It defaults to "off" for open source projects, meaning that a malicious user can't simply print the secrets in a fork. In other words, you can only leak the secrets after a PR has been merged, which should only happen after careful review.

For other projects (e.g. HarbourBridge), we have set up both unit tests and integration tests. The unit tests always run, even for forked pull requests (using the "Build forked pull requests" setting, which currently is off for wrench). However, we skip the integration tests when the corresponding environment variables aren't set, i.e. we keep the "Pass secrets to builds from forked pull requests" setting turned off. The integration tests will then only run after a PR has been merged, which seems a good compromise between presubmit testing and full coverage through integration tests.

Reply:
Thank you for precisely explaining!

> The integration tests will then only run after a PR has been merged, which seems a good compromise between presubmit testing and full coverage through integration tests.

I agreed this strategy!